src/firefly/js/metaConvert/vo/ObsCoreConverter.js

```diff
@@ -201,10 +201,10 @@ async function getObsCoreSingleDataProduct({table, row, activateParams, serviceD
 
 
         const positionWP= getSearchTarget(table.request,table) ?? makeWorldPtUsingCenterColumns(table,row);
-        const request= makeObsCoreRequest(dataSource, positionWP, titleStr,table,row);
+        const request= makeObsCoreRequest(dataSource, cloudAccess, positionWP, titleStr,table,row);
         const primDPType= doFileAnalysis ?
             await uploadAndAnalyze({request,table,row,activateParams,serviceDescMenuList,originalTitle:request.getTitle()}) :
-            createGuessDataType(titleStr,'guess-0',dataSource,prodType,undefined,activateParams, positionWP,table,row,size);
+            createGuessDataType(titleStr,'guess-0',dataSource,cloudAccess, prodType,undefined,activateParams, positionWP,table,row,size);
```

Comment:
It might be a bit late, but these functions really should have taken the entire data structure from getObsCoreRowMetaInfo() as a single parameter. Right now, there are too many repeating parameters without any grouping.

Reply:
Just FYI: makeObsCoreRequest parameters come from more that a obscore table row. It can also be data link and some of the parameters are computed. The table and row parameters might be an obscore table with the access url or the reference obscore table when the access_url is from datalink. Or the dataSource might be computed in a service descriptor. (in the case cloud_access would be undefined).